### ISSUE TITLE
NestedClassesVisibility false positive

### ISSUE THREAD

**Issue by igorwojda** [arturbosch/detekt#1590]
`NestedClassesVisibility` should take into consideration `sealed` keyword.
Now it would complain that rule is violated because subclasses of `internal sealed MyTest` class are `public`, however, they are still not accessible outside the module (It looks like kotlin compiler makes them `internal`).

## Your Environment
Module A
```
internal sealed class MyTest { //NestedClassesVisibility violation
    object MyTest2: MyTest()
    class MyTest3(): MyTest()
} 
```

Module B
```
// In reality, can't access class outside Module A, so there is no violation
val foo: MyTest2 //can't access error 
```

**Comment by schalkms** [comment]
I just stumbled upon this issue.

This issue has nothing to do with the `sealed` keyword and the `NestedClassesVisibility` rule. 
Because your class `MyTest` is `internal`, you cannot access it in another module no matter whether it's `sealed` or not.
Thus, this code is no false-positive.

I don't see any reason for changing the rule here.

**Comment by schalkms** [comment]
> Now it would complain that rule is violated because subclasses of `internal sealed MyTest` class are `public`, however, they are still not accessible outside the module (It looks like kotlin compiler makes them `internal`).

@igorwojda 
Do you have any reference for your assumption, that the Kotlin compiler makes the nested classes `internal`?

**Comment by igorwojda** [comment]
You are right. I am closing the issue.
